ui/src/components/LoginForm/SelectProposal.jsx

```diff
@@ -1,182 +1,123 @@
-import React from 'react';
-import { connect } from 'react-redux';
-import { reduxForm } from 'redux-form';
+import React, { useState } from 'react';
 import { Modal, Button, Tabs, Tab, Form } from 'react-bootstrap';
 import SessionTable from './SessionTable';
-
-class SelectProposal extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.handleSelectProposal = this.handleSelectProposal.bind(this);
-    this.handleCancel = this.handleCancel.bind(this);
-    this.handleOnSessionSelected = this.handleOnSessionSelected.bind(this);
-
-    this.state = {
-      pId: 0,
-      pNumber: null,
-      session: null,
-      proposal: null,
-      filter: null,
-      sessions: props.data.proposalList,
-      filteredSessions: props.data.proposalList,
-    };
```

Comment:
Note that I've simplified the state a fair bit: I now have only a `filter` state and a `selectedSession` state. The `sessions` don't need to be stored in the state, and the rest can just be computed on every render.